http/query/query.go

```diff
@@ -36,16 +36,15 @@ func New(query string) (*Query, error) {
 
 	for i, length := 0, len(query); i < length; i++ {
 		separator := query[i] == '&' || query[i] == ';' || query[i] == '?'
-		if separator || i == length-1 {
-			if onKey {
+		last := i == length-1
+
+		if separator || last {
+			if onKey && !last {
 				keyStart = i + 1
 				continue
 			}
 
-			if i == length-1 && !separator {
-				if query[i] == '=' {
-					continue
-				}
```

Comment:
This existed to prevent queries like `foo=bar&baz=` from generating a blank `baz` key, but we actually do want that.